lib/utils/computed-property-dependent-keys.js

```diff
@@ -256,15 +249,65 @@ function findComputedPropertyDependentKeys(nodeClass, computedImportName, macroI
  * @param {number} dependentKeyArgumentCount - number of arguments to check for dependent keys
  * @returns {String[]} - the list of string dependent keys from this computed property
  */
-function getComputedPropertyDependentKeys(node, dependentKeyArgumentCount = Number.MAX_VALUE) {
+function getComputedPropertyDependentKeys(node, computedDependenciesConfig = null) {
   if (!node.arguments) {
     return [];
   }
 
-  return expandKeys(
-    node.arguments
-      .slice(0, dependentKeyArgumentCount)
-      .filter((arg) => arg.type === 'Literal' && typeof arg.value === 'string')
-      .map((node) => node.value)
-  );
+  const baseKeys = javascriptUtils.flatMap(node.arguments, (argument, index) => {
+    if (!computedDependenciesConfig) {
+      if (argument.type === 'Literal' && typeof argument.value === 'string') {
+        return [argument.value];
+      } else {
+        return [];
+      }
+    }
+
+    return javascriptUtils.flatMap(
+      computedDependenciesConfig.argumentFormat,
+      ({ strings, objects }) => {
```

Comment:
This callback function is very complicated, can it be extracted into its own function with some comments?